bad_code.java

```diff
@@ -0,0 +1,54 @@
+public class BadCodeExample {
+    
+    public static void main(String[] args) {
+        ChildClass child = new ChildClass();
+        child.doSomething();
+        child.doSomethingElse();
+    }
+}
+
+class ParentClass {
+    public int someNumber = 10;
+    public String someString = "Hello";
+
+    public void doSomething() {
+        System.out.println("Doing something in ParentClass");
+        someNumber++;
+        someString += " Worlds";
+        System.out.println(someString);
+    }
+
+    public void doSomethingElse() {
+        for (int i = 0; i < someNumber; i++) {
```

Comment:
In the `doSomethingElse` method of `ChildClass`, consider using a more efficient way to iterate over the range, such as using a `for-each` loop or a `Stream` API. [medium]